examples/api/lib/widgets/animated_list/animated_list_separated.0.dart

```diff
@@ -0,0 +1,278 @@
+
+  @override
+  State<AnimatedListSeparatedSample> createState() => _AnimatedListSeparatedSampleState();
+}
+
+class _AnimatedListSeparatedSampleState extends State<AnimatedListSeparatedSample> {
+  final GlobalKey<AnimatedListState> _listKey = GlobalKey<AnimatedListState>();
+  late ListModel<int> _list;
+  int? _selectedItem;
+  late int _nextItem; // The next item inserted when the user presses the '+' button.
+
+  @override
+  void initState() {
+    super.initState();
+    _list = ListModel<int>(
+      listKey: _listKey,
+      initialItems: <int>[0, 1, 2],
+      removedItemBuilder: _buildRemovedItem,
+    );
+    _nextItem = 3;
+  }
+
+  // Used to build list items that haven't been removed.
+  Widget _buildItem(BuildContext context, int index, Animation<double> animation) {
+    return CardItem(
+      animation: animation,
+      item: _list[index],
+      selected: _selectedItem == _list[index],
+      onTap: () {
+        setState(() {
+          _selectedItem = _selectedItem == _list[index] ? null : _list[index];
+        });
+      },
+    );
+  }
+
+  // Used to build separators for items that haven't been removed.
+  Widget _buildSeparator(BuildContext context, int index, Animation<double> animation) {
+    return ItemSeparator(
+      animation: animation,
+      item: _list[index],
+    );
+  }
+
+  /// The builder function used to build items that have been removed.
+  ///
+  /// Used to build an item after it has been removed from the list. This method
+  /// is needed because a removed item remains visible until its animation has
+  /// completed (even though it's gone as far as this ListModel is concerned).
+  /// The widget will be used by the [AnimatedListState.removeItem] method's
+  /// `itemBuilder` parameter.
+  Widget _buildRemovedItem(int item, BuildContext context, Animation<double> animation) {
+    return CardItem(
+      animation: animation,
+      item: item,
+      // No gesture detector here: we don't want removed items to be interactive.
+    );
+  }
+
+  /// The builder function used to build a separator for an item that has been removed.
+  ///
+  /// Used to build a separator after the corresponding item has been removed from the list.
+  /// This method is needed because the separator of a removed item remains visible until its animation has completed.
+  /// The widget will be passed to [AnimatedList.separated]
+  /// via the [AnimatedList.removedSeparatorBuilder] parameter and used
+  /// in the [AnimatedListState.removeItem] method.
+  ///
+  /// The item parameter is null, because the corresponding item will
+  /// have been removed from the list model by the time this builder is called.
+  Widget _buildRemovedSeparator(BuildContext context, int index, Animation<double> animation) {
+    return SizeTransition(
+                sizeFactor: animation,
+                child: ItemSeparator(
+                  animation: animation,
+                  item: null,
+                )
+              );
```

Comment:
The indentation here is off, can you fix it?

Reply:
fixed :)